tests/scale/README.md

```diff
@@ -0,0 +1,154 @@
+- It should also provide instrumentation to gate at PR level (possibly an on-demand pipeline), in order to find possible performance/scale regressions before merge.
+- It should also provide meaningful insight on how the performance and scalability of the overall product offering evolves at each milestone/release.
+
+## Design
+
+Current tests leverage the existing test framework to deploy OSM and repeat a work operation (iteration) till the validation phase of the iteration fails. How and what does the iteration do is up to the test, so implementation is free to scale any resource/s till failure.
+
+The framework provides helpers to track profiling information through the overall test and individual iterations. Some of most relevant metrics are automatically captured during the test, but more exclusive metrics for speicific tests might have to be implemented. 
+Gathering these metrics require a Prometheus instance scraping both the envoy endpoints and the K8s api servers.
+
+The current set of metrics aimed to be automatically tracked across iteration for any set of resources specified in the test are:
+- CPU Average loads, for each iteration for each tracked resource.
+- RSS footprint and related relative increases per iteration per tracked resource.
+- Visual representation of the previous trends, provided by Grafana.
+- Control plane profiling (pprof), cpu and mem (Todo)
+- Envoy config latency trends (time to create an envoy config by osm-controller, latency increase per pod, test dependent) (Todo)
+- Envoy config latency apply trends (from  `SMI apply` to `200` network requests) (Todo)
+
+
+## Usage examples
+Create a test using GinkGo, and import the test framework. As we do for our E2E, we require a separate test per file.
+```go
+import . "github.com/openservicemesh/osm/tests/framework"
+
+var  _ = Describe("Example Skeleton for scale test", func() {
+	Context("ScaleTest", func() {
+		// sd to store test state and iteration information
+		var  sd *DataHandle
+
+		// WrapUp will call to evaluate and output results on test's on os.Stdout.
+		// Additional outputs can be defined through `sd` API
+	AfterEach(func() {
+		if sd != nil {
+			sd.WrapUp()
+		}
+	})
+})
+```
+
+Initialize the body of the test, pretty standard from E2E framework semantics:
+```go
+It("Deploys OSM and scales traffic Splits indefinitely", func() {
+	// Install OSM, enable Grafana and Prometheus self-deployment
+	t := Td.GetOSMInstallOpts()
+	t.DeployGrafana = true
+	t.DeployPrometheus = true
+	Td.InstallOSM(t)
+
+	// Helpers to get OSM's install handlers, but arbitrary ones can be provided
+	pHandle := Td.GetOSMPrometheusaHandle()
+	gHandle := Td.GetOSMGrafanaHandle()
+
+	// This could/should be called on `Context`, subject to when are resources available
+	sd = NewDataHandle(pHandle, gHandle, GetTrackedResources(), GetSaveDashboards())	
+
+	// Repetitive scale loop
+	sd.Iterate(func() {
+		// Code goes here
```

Comment:
Is each test responsible for defining exactly how the scaling works between iterations? I'm not seeing anything in the helpers that's creating more pods.

Reply:
yeah each tests defines what scaling means after all. Some tests might not be interested scaling pods, but policies for example.